### PR TITLE
fix: fix tabs at beginning of list items

### LINKED ISSUE #2516: Breaking change of leading tab in unordered list
**What pain point are you perceiving?.**
When using 2 leading tabs in a list it starts seeing it as a code block. Which could make sense
But starting from 4.0.15 it also places them on another level which I really can't explain.

Marked has introduced this breaking change between version 4.0.13 & 4.0.14, and 4.0.15 makes it even worse.
Caused by #2434 

[Working example 4.0.13](https://marked.js.org/demo/?text=%23%23%23%20%E2%9A%A0%EF%B8%8F%20Hello%20World%20%E2%9A%A0%EF%B8%8F%0A%0A-%20A%0A-%20%20%20B%0A-%09C%0A-%09D%0A-%09E%0A-%09F&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=4.0.13)
[Broken example 4.0.14](https://marked.js.org/demo/?text=%23%23%23%20%E2%9A%A0%EF%B8%8F%20Hello%20World%20%E2%9A%A0%EF%B8%8F%0A%0A-%20A%0A-%20%20%20B%0A-%09C%0A-%09D%0A-%09E%0A-%09F&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=4.0.14)
[Even more broken example 4.0.15](https://marked.js.org/demo/?text=%23%23%23%20%E2%9A%A0%EF%B8%8F%20Hello%20World%20%E2%9A%A0%EF%B8%8F%0A%0A-%20A%0A-%20%20%20B%0A-%09C%0A-%09D%0A-%09E%0A-%09F&options=%7B%0A%20%22baseUrl%22%3A%20null%2C%0A%20%22breaks%22%3A%20false%2C%0A%20%22extensions%22%3A%20null%2C%0A%20%22gfm%22%3A%20true%2C%0A%20%22headerIds%22%3A%20true%2C%0A%20%22headerPrefix%22%3A%20%22%22%2C%0A%20%22highlight%22%3A%20null%2C%0A%20%22langPrefix%22%3A%20%22language-%22%2C%0A%20%22mangle%22%3A%20true%2C%0A%20%22pedantic%22%3A%20false%2C%0A%20%22sanitize%22%3A%20false%2C%0A%20%22sanitizer%22%3A%20null%2C%0A%20%22silent%22%3A%20false%2C%0A%20%22smartLists%22%3A%20false%2C%0A%20%22smartypants%22%3A%20false%2C%0A%20%22tokenizer%22%3A%20null%2C%0A%20%22walkTokens%22%3A%20null%2C%0A%20%22xhtml%22%3A%20false%0A%7D&version=4.0.15)

**Describe the solution you'd like**
A solution where it keeps working as it used to. I checked GitHub and they seem to ignore the tabs.
[CommonMark](https://spec.commonmark.org/dingus/?text=%23%23%23%20%E2%9A%A0%EF%B8%8F%20Hello%20World%20%E2%9A%A0%EF%B8%8F%0A%0A-%20A%0A-%20%20%20B%0A-%09C%0A-%09D%0A-%09E%0A-%09F) seems to agree this is wrong. Dillinger also ignores it.

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** 4.2.4

## Description

- Fixes #2516 

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

File: src/Tokenizer.js [hunk 1]
```diff
@@ -203,7 +203,7 @@ export class Tokenizer {
         raw = cap[0];
         src = src.substring(raw.length);
 
-        line = cap[2].split('\n', 1)[0];
+        line = cap[2].split('\n', 1)[0].replace(/^\t+/, (t) => ' '.repeat(3 * t.length));
```
Comment (styfle) [inline src/Tokenizer.js:206]: Whats the significance of `3` here?

Comment (UziTech) [reply src/Tokenizer.js:206]: That is how common mark does it

**Comment by calculuschild** [comment]
I'm a little wary of how much unique handling of tabs is getting buried in the Lists Tokenizer recently. I'm not sure where, but somehow it seems like some of this stuff should be transformed ahead of time in a universal way so other tokens get the same tab behavior. Does any of that apply here?

**Comment by UziTech** [comment]
If we try to handle tabs like this we need to make sure it is actually the start of a list item. I think it would be a lot slower to check for list twice to transform the tabs somewhere else.

It would be really nice if we could actually handle tabs correctly instead of replacing them with spaces but I feel like that would be a lot of work (and probably slower).

**Review by styfle — APPROVED** [review]
Approving just for the sake of having the test and then we can revise the implementation later

**Pull request merged**